zephyr/src/rt/CMakeLists.txt

```
@@ -0,0 +1,71 @@
+if(CONFIG_NXP_MONOLITHIC_WIFI OR CONFIG_BT_H4_NXP_CTLR)
+	set(hal_blobs_dir ${ZEPHYR_HAL_NXP_MODULE_DIR}/zephyr/blobs)
+	set(wifi_bt_binary_blobs_list)
+	set(binary_blobs_list)
+	set(output_includes_list)
+
+	if(CONFIG_WIFI_NXP)
+            if(CONFIG_NXP_IW61X)
+                list(APPEND output_includes_list ${ZEPHYR_BINARY_DIR}/include/generated/nw61x_wifi_fw.bin.se.inc)
+                set(wifi_binary_blob_name sduart_nw61x.bin.se)
+                zephyr_library_sources(${CMAKE_CURRENT_LIST_DIR}/nw61x/nw61x_cpu1.c)
+                list(APPEND wifi_bt_binary_blobs_list ${hal_blobs_dir}/nw61x/${wifi_binary_blob_name})
+            elseif(CONFIG_NXP_88W8987)
+                list(APPEND output_includes_list ${ZEPHYR_BINARY_DIR}/include/generated/8987_wifi_fw.bin.inc)
+                set(wifi_binary_blob_name sd8987_wlan.bin)
+                zephyr_library_sources(${CMAKE_CURRENT_LIST_DIR}/8987/8987_cpu1.c)
+                list(APPEND wifi_bt_binary_blobs_list ${hal_blobs_dir}/8987/${wifi_binary_blob_name})
+            elseif(CONFIG_NXP_IW416)
+                list(APPEND output_includes_list ${ZEPHYR_BINARY_DIR}/include/generated/iw416_wifi_fw.bin.inc)
+                set(wifi_binary_blob_name sdIW416_wlan.bin)
+                zephyr_library_sources(${CMAKE_CURRENT_LIST_DIR}/IW416/IW416_cpu1.c)
+                list(APPEND wifi_bt_binary_blobs_list ${hal_blobs_dir}/IW416/${wifi_binary_blob_name})
+            elseif(CONFIG_NXP_88W8801)
+                list(APPEND output_includes_list ${ZEPHYR_BINARY_DIR}/include/generated/8801_wifi_fw.bin.inc)
+                set(wifi_binary_blob_name sd8801_wlan.bin)
+                zephyr_library_sources(${CMAKE_CURRENT_LIST_DIR}/8801/8801_cpu1.c)
+                list(APPEND wifi_bt_binary_blobs_list ${hal_blobs_dir}/8801/${wifi_binary_blob_name})
+            else()
+                message(FATAL_ERROR "Couldn't determine soc revision")
+            endif()
+	endif()
+
+	if (CONFIG_BT_H4_NXP_CTLR)
+		if(CONFIG_BT_NXP_NW612)
```

We need to consider the case that if the configuration `CONFIG_BUILD_ONLY_NO_BLOBS` is true, ignore the any controller, and give a simulated blobs. It is just used to pass the building.
```suggestion
		if(CONFIG_BUILD_ONLY_NO_BLOBS)
			set(bt_binary_blob_name bt_no_blobs.bin)
			zephyr_library_sources(${CMAKE_CURRENT_LIST_DIR}/bt_no_blobs.c)
		elseif(CONFIG_BT_NXP_NW612)
```

And, in file `${CMAKE_CURRENT_LIST_DIR}/bt_no_blobs.c`, my suggestion likes as,
```
/*
 * Copyright 2025 NXP
 * All rights reserved.
 *
 * SPDX-License-Identifier: BSD-3-Clause
 *
 */

#include <stdint.h>

const unsigned char *bt_fw_bin = NULL;
const unsigned int bt_fw_bin_len = 0;
```
The configuration `CONFIG_BUILD_ONLY_NO_BLOBS` is required by Bluetooth now. I am not sure whether it is also required by WIFI.

---

Hi @lylezhu2012 , NO BLOB support is added, blobs verify support is also added in this PR. thank you